### PR TITLE
Ft/lim 54

### PR DESCRIPTION
**Pull request by Bolaji06**
Wrote test suits for categories endpoint for all the possible edge cases I can think of...

### CODE REVIEW

File: test/category.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    it("should return 401 if limit is greater than 20", async () => {
+        const response = await request(app).get('/api/categories').query({ limit: 24 });
+        expect(response.statusCode).toBe(401);
+        expect(response.body).toEqual({ success: false, message: "you can only request 20 records per page" });
+    })
+
+    it("should return categories with default pagination values", async () => {
+        const mockCategories = [{ category: "Fiction" }, {category: "Non-Fiction" }];
+
+        prisma.category.findMany = jest.fn().mockResolvedValue(mockCategories)
```
Comment (Bolaji06) [inline test/category.test.js:42]: Do you have problem using prisma mockResolvedValue?

File: test/category.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        expect(prisma.category.findMany).toHaveBeenCalledWith({
+            skip: 0,
+            take: 20,
+        })
+    });
+
+    it("should 200 with a number of page and number of limit", async () => {
+        const mockCategories = [{ category: "Adventure"}, { category: "Science" }, { category: "Literature" }];
+
+        prisma.category.findMany = jest.fn().mockResolvedValue(mockCategories);
```
Comment (Bolaji06) [inline test/category.test.js:56]: Same here??

**Review by Bolaji06 — COMMENTED** [review]
Nice work. will be merging

File: test/category.test.js (hunk 1 above)
Comment (Bolaji06) [reply test/category.test.js:42]: I was getting can't resolve mockResolvedValue error.
